pkg/ddl/ddl.go

```diff
@@ -416,20 +449,51 @@ func (sv *schemaVersionManager) setSchemaVersion(job *model.Job, store kv.Storag
 }
 
 // lockSchemaVersion gets the lock to prevent the schema version from being updated.
-func (sv *schemaVersionManager) lockSchemaVersion(jobID int64) {
+func (sv *schemaVersionManager) lockSchemaVersion(jobID int64) error {
 	ownerID := sv.lockOwner.Load()
 	// There may exist one job update schema version many times in multiple-schema-change, so we do not lock here again
 	// if they are the same job.
 	if ownerID != jobID {
 		sv.schemaVersionMu.Lock()
 		sv.lockOwner.Store(jobID)
+		if sv.etcdClient != nil && variable.DDLVersion.Load() == model.TiDBDDLV2 {
+			se, err := concurrency.NewSession(sv.etcdClient)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			mu := concurrency.NewMutex(se, ddlSchemaVersionKey)
+			if err := mu.Lock(sv.ctx); err != nil {
+				return errors.Trace(err)
+			}
+			sv.seMaps[jobID] = se
+			sv.lockMaps[jobID] = mu
+		}
 	}
+	return nil
 }
 
 // unlockSchemaVersion releases the lock.
 func (sv *schemaVersionManager) unlockSchemaVersion(jobID int64) {
 	ownerID := sv.lockOwner.Load()
 	if ownerID == jobID {
+		if sv.etcdClient != nil && variable.DDLVersion.Load() == model.TiDBDDLV2 {
```

Comment:
What happen if a job is marked with `local_mode`, but `variable.DDLVersion` is `TiDBDDLV1`?

Reply:
https://github.com/pingcap/tidb/blob/e5195d09d6948f0f3db14160e6e9f0b0da48e94e/pkg/ddl/ddl.go#L1419

Reply:
Could we use `if se, ok := sv.seMaps[jobID]; ok` to check if we need to do `mu.Unlock`? Then we needn't to dependent the varible.